keps/sig-storage/20191125-bucket-provisioning.md

```diff
@@ -0,0 +1,274 @@
+---
+title: Object Bucket Provisioning
+authors:
+  - "@jeffvance"
+  - "@copejon"
+owning-sig: "sig-storage"
+reviewers:
+  - "@saad-ali"
+  - "@alarge"
+  - "@erinboyd"
+  - "@guymguym"
+  - "@travisn"
+approvers:
+  - TBD
+editor: TBD
+creation-date: 2019-11-25
+last-updated: 2020-02-27
+status: provisional
+---
+
+# Object Bucket Provisioning
+
+## Table of Contents
+
+<!-- toc -->
+- [Summary](#summary)
+  - [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+  - [Vocabulary](#vocabulary)
+- [Proposal](#proposal)
+  - [User Stories](#user-stories)
+      - [Admin](#admin)
+      - [User](#user)
+  - [System Configuration](#system-configuration)
+  - [Workflows](#workflows)
+    - [Create Bucket](#create-bucket)
+    - [Delete Bucket](#delete-bucket)
+    - [Grant Bucket Access](#grant-bucket-access)
+    - [Revoke Bucket Access](#revoke-bucket-access)
+  - [Custom Resource Definitions](#custom-resource-definitions)
+      - [ObjectBucketClaim (OBC)](#objectbucketclaim-obc)
+      - [ObjectBucket (OB)](#objectbucket-ob)
+      - [BucketClass](#bucketclass)
+<!-- /toc -->
+
+# Summary
+
+This proposal introduces the Container Object Storage Interface (COSI), whose purpose is to provide a familiar and standardized method of provisioning object storage buckets in an manner agnostic to the storage vendor. The COSI environment is comprised of Kubernetes CRDs, operators to manage these CRDs, and a gRPC interface by which these operators communicate with vendor plugins.  This design is of course heavily inspired by the Kubernetes’ implementation of the Container Storage Interface (CSI).  However, bucket management lacks some of the notable requirements of block and file provisioning and allows for an architecture with lower overall complexity.  This proposal does not include a standardized protocol or abstraction of storage vendor APIs.  
```

Comment:
It has nothing to do with Containers. So I think Container Object Storage Interface may not be the right term.